app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java

```diff
@@ -444,8 +444,16 @@ public void filesPopulated() {
             checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
                 if (isChecked) {
                     if (!mSelectedFiles.contains(getAdapterPosition())) {
-                        mSelectedFiles.add(getAdapterPosition());
-                        itemSelectedListener.isSelected(true, mSelectedFiles.size());
+                        String mFileName = mFileList.get(getAdapterPosition()).getPdfFile().getPath();
+                        if (mPDFEncryptionUtils.isPDFEncrypted(mFileName)) {
+                            mPDFEncryptionUtils.removePassword(mFileName,
+                                    ViewFilesAdapter.this);
+                            mSelectedFiles.add(getAdapterPosition() + 1);
+                            itemSelectedListener.isSelected(true, mSelectedFiles.size());
```

Comment:
These lines will check if pdf is encrypted, ask user for password and if it's all good - insert an entry for unencrypted file in the adapter?

Reply:
Don't we want to remove the password protected file from list?

Reply:
> These lines will check if pdf is encrypted, ask user for password and if it's all good - insert an entry for unencrypted file in the adapter?

Yes

Reply:
> Don't we want to remove the password protected file from list?

Oh yeah sorry forgot about it.!

Reply:
> Don't we want to remove the password protected file from list?

In the program, the password protected file is not added before decryption. So there is no need to remove the encrypted file.

Reply:
Sorry for the misunderstanding, but I assumed we were implementing this for "Merge Files" section.

We don't want users to type in the password just for selecting the files. This will be hampering our normal View Files section behvaiour.

Reply:
Yeah I checked even for Merge PDF section the encrypted files are not being merged. So what do we do now? Shall I implement this feature to "Merge PDF" section instead of view files section?

Reply:
You were correct initiallly.
- For View Files section, we should show repeated dialogs for encrypted PDFs after the user has selected PDF
- For Merge Files section, we can go with current approach -> ask for password when file is selected.

You can pick up either of the issue. Thanks!

Reply:
Sure @Swati4star I'll go with the second issue. Thank you!

Reply:
@Swati4star I'm confused how to ask for passwords when user is in the "internal files selection" intent. I mean when the user selects a file from local how can I ask a password for it from the app?

Reply:
After the user has selected the file, and is back in our app 
Somewhere inside this `if` (https://github.com/Swati4star/Images-to-PDF/blob/master/app/src/main/java/swati4star/createpdf/fragment/MergeFilesFragment.java#L254), we can check if file is encrypted and ask them for password

Reply:
Ok I'll try this...

Reply:
When I try to get the path of encrypted file it is null!. What to do in that case?

Reply:
Can you debug why is it coming out to be null?

Ideally, if you haven't pressed back button, the path should be non-null. There must be a bug in our code if we are getting null for valid paths.

Reply:
The null value nice resolved it but one issue is that the path of the file needs to be changed to path_unencrypted.pdf instead of path.pdf. but the program is not waiting for the path to be renamed. Because the renamed file is supposed to be merged right.

Reply:
we can get the original PDF selected from user, and then ask for password and decrypt the PDf, and then add new PDF to the list of files to be merged. (and remove previous one)